src/content/docs/en/reference/modules/astro-middleware.mdx

```diff
@@ -16,60 +16,84 @@ Middleware allows you to intercept requests and responses and inject behaviors d
 
 ## Imports from `astro:middleware`
 
+The following helpers are imported from the virtual middleware module:
+
 ```js
 import { 
-  sequence,
-  createContext,
-  trySerializeLocals,
   defineMiddleware,
- } from 'astro:middleware';
+  sequence,
+} from 'astro:middleware';
 ```
 
 ### `defineMiddleware()`
 
-You can import and use the utility function `defineMiddleware()` to take advantage of type safety:
+<p>
+
+**Type:** <code>(fn: <a href="#middlewarehandler">MiddlewareHandler</a>) => MiddlewareHandler</code>
+</p>
+
+A function for defining a middleware function with type safety. When you use this utility, the [`context`](#context) and [`next()`](#next) arguments are automatically typed, and you will get a Typescript error if you try to return a [value not supported](#middlewarehandler) in your middleware.
```

Comment:
```suggestion
A function for defining a middleware function with type safety. When you use this utility, the [`context`](#context) and [`next()`](#next) arguments are automatically typed, and you will get a Typescript error if you try to return a [value not supported in your middleware](#middlewarehandler).
```
I was trying to think of the best link text here. I don't think "value not supported" on its own is the *best* if it points to the middlewarehandler. It could be all of this, maybe? Thoughts?